common/types/activation.go

```diff
@@ -299,18 +241,28 @@ func (atx *ActivationTx) MarshalLogObject(encoder log.ObjectEncoder) error {
 	return nil
 }
 
-// CalcAndSetID calculates and sets the cached ID field. This field must be set before calling the ID() method.
-func (atx *ActivationTx) CalcAndSetID() error {
+// Initialize calculates and sets the cached ID field. This field must be set before calling the ID() method.
+func (atx *ActivationTx) Initialize() error {
+	if atx.ID() != EmptyATXID {
+		return fmt.Errorf("ATX already initialized")
```

Comment:
either panic or return nil.
this is a programming error.

Reply:
Ballot and blocks also behave this way? ATX is the only odd one out that panics instead if returning an error if initialize is called twice.

Reply:
for ballot/proposal/block, Initialize() is also called on the gossip handler path. ok. i see now you also use this in atx gossip handler path.